File: golang/array_string/removeelement.go
package main

func removeElement(nums []int, val int) int {
	slow := 0

	for fast := 0; fast < len(nums); fast++ {
		if nums[fast] != val {
			nums[slow] = nums[fast]

			slow += 1
		}
	}

	return slow
}

File: java/src/main/java/com/sanedge/interview/array_string/RemoveElement.java
package com.sanedge.interview.array_string;

public class RemoveElement {

    public int removeElement(int[] nums, int val) {
        int slow = 0;

        for (int fast = 0; fast < nums.length; fast++) {
            if (nums[fast] != val) {
                nums[slow] = nums[fast];

                slow += 1;
            }
        }

        return slow;
    }
}
